### DISCUSSION TITLE
After update can no longer stage hunks by visual selection

### DISCUSSION THREAD

**Discussion by dam5h** [magit/magit#4685] [Support] [answered]
One of the workflows (among many) that I enjoy using magit for is using `magit-diff` to show either all unstaged changes or all staged changes.  I could then press "v" to visually select lines and use "u" or "s" to unstage (if staged) or stage (if unstaged) respectively.  I can no longer do this, I can still visually select lines in the diff and "discard" them with "x" but cannot stage them with "s".  

Has this workflow been deprecated or changed in some way, if so what has it been replaced with?

EDIT: As a follow up, I can use "s" when staying the the magit buffer and not viewing the "diff". If I expand the changes by file via the tab key in the main magit buffer then I can stage by visual selection. What I am missing is doing it from the diff view.

I am using spacemacs and magit version 3.3.0-git.

Thanks!

**Comment by tarsius** [comment, marked as answer]
> Has this workflow been deprecated

Of course not!

> or changed in some way, if so what has it been replaced with?

No.

> but cannot stage them with "s"

What happens if you try? Are there any error messages?

Did you try to reproduce with `emacs -Q`? See https://magit.vc/manual/magit/Debugging-Tools.html.

**Reply by dam5h** [reply]
I did not try to reproduce with `emacs -Q`. I just tried that but it doesn't load anything.  It just gives me a plain vanilla emacs terminal that doesn't have any magit commands.  I have to admit to not being very good in vanilla emacs!  I did try `M-x mag...` and it has none of the commands.

When I try typing the `s` I get no errors that I know of but an `s` appears in my status line at the bottom of the buffer.  If I press `s` again I get a `Buffer is read only: #<buffer magif-diff: X>` message in the status line, since it thinks I want to start editing.  I use `evil` mode via spacemacs so I think it is interpretting the `s` as replace at point and start editing.

Maybe this is useful, I do see an error that came up once, but not when I hit the `s` key:

```
Cannot insert ("N" "Forge" forge-dispatch) into magit-dispatch; o not found
Setting 'forge-add-default-bindings' to nil in 'evil-collection-forge-setup'.
```
Could that be related?  I do not get this message on my desktop that I have not updated and stages fine.

Thanks for all the help.  And thank you very much for magit!!

**Reply by tarsius** [reply]
Sounds like an issue with spacemacs/evil key bindings. I can't really help you with that.

**Reply by dam5h** [reply]
Cool, that makes sense.  I'll report back when I have time to figure it out.  Thanks again, magit is :fire: !
